velox/exec/fuzzer/JoinFuzzer.cpp

```diff
@@ -606,66 +606,103 @@ std::optional<core::JoinType> tryFlipJoinType(core::JoinType joinType) {
 }
 
 // Returns a plan with flipped join sides of the input hash join node. If the
-// join type doesn't allow flipping, returns a nullptr.
+// inputs of the join node are other hash join nodes, recursively flip the join
+// sides of those join nodes as well. If the join type doesn't allow flipping,
+// returns a nullptr.
 core::PlanNodePtr tryFlipJoinSides(const core::HashJoinNode& joinNode) {
   //  Null-aware right semi project join doesn't support filter.
   if (joinNode.filter() &&
       joinNode.joinType() == core::JoinType::kLeftSemiProject &&
       joinNode.isNullAware()) {
     return nullptr;
   }
+  core::PlanNodePtr left = joinNode.sources()[0];
+  core::PlanNodePtr right = joinNode.sources()[1];
+  if (auto leftJoinInput = std::dynamic_pointer_cast<const core::HashJoinNode>(
```

Comment:
Shouldnt both leftJoinInput and rightJoinInput succeed for us to try and flip join sides ?

Reply:
No. The sources of the join can either be another hash join node or a non-join node (e.g values node). If a source is another hash join node, then it's join sides should also be flipped.